assignment/assignment-tool/tool/src/webapp/vm/assignment/assignment_macros.vm

```diff
@@ -37,12 +37,12 @@
 ## Defining macro to repeat buttons at the top and bottom of the page
 #macro( buttonBar $submitnotif )
     <div class="act">
-        <input accesskey="s" type="button" class="active" name="post" value="$tlang.getString("gen.pos")" onclick="SPNR.disableControlsAndSpin( this, null ); ASN.submitForm( 'newAssignmentForm', 'post', null, null, true );" />
-        <input accesskey="v" type="button" name="preview" value="$tlang.getString("gen.pre")" onclick="SPNR.disableControlsAndSpin( this, null ); ASN.submitForm( 'newAssignmentForm', 'preview', null, null, true );" />
+        <input accesskey="s" type="button" class="active" id="post" name="post" value="$tlang.getString("gen.pos")" onclick="SPNR.disableControlsAndSpin( this, null ); ASN.submitForm( 'newAssignmentForm', 'post', null, null, true );" />
```

Comment:
Technincally, @buckett is correct. The buttonBar is used twice on this page, once at the top and once at the bottom. In terms of compliance, they should have unique IDs.

In terms of functionality, it still works with this patch, and the spinner will only be applied to the button you actually clicked on because it's passed to the spinner algorithm via `this`.

However, we should probably just take the extra few minutes to ensure unique IDs. An easy way is to just add another parameter to the macro definition, such as:

`#macro( buttonBar $submitnotif $instanceNum )`

And then append the `$instanceNum` to the id of the element:

`id="post$instanceNum"`

The responsibility will then fall on the programmer to remember to supply a unique instance number in the calling location:

```
#buttonBar( #submitnotif 1 )
...
...
#buttonBar( #submitnotif 2 )
```